src/Generalize.js

```diff
@@ -44,11 +44,49 @@ L.MarkerGeneralizeGroup = L.FeatureGroup.extend({
 
         this._zoomReady = {};
 
+        this.setMaxZoom(options.maxZoom);
+        this.setMinZoom(options.minZoom);
+
         this.on('invalidationFinish', function() {
             this._map.getPanes().markerPane.style.display = 'block';
         });
     },
 
+    setMaxZoom: function(zoom) {
+        if (isNaN(zoom) || zoom > 19) {
+            this._maxZoom = 19;
+            this._userMaxZoom = false;
```

Comment:
Looks like we can do the same without `_userMaxZoom`. Just don't assign it if it is not passed, and then use `_maxZoom || map.getMaxZoom()` or something like this.